### PR TITLE
CA-155178: Remove squeeze_xen cache if domain disappear

### PR DESCRIPTION
**Pull request by cheng-z**
Cleanup disappear domains in domain make to avoid dirty domain cache

Signed-off-by: Cheng Zhang <cheng.zhang@citrix.com>

### CODE REVIEW

**Comment by cheng-z** [comment]
Hi @johnelse 
Could you please help to check if this code can be merge?
The related pull request is:
https://github.com/xapi-project/squeezed/pull/32/files

File: src/squeeze_xen.ml [hunk 1]
```diff
@@ -86,6 +86,22 @@ module Domain = struct
 			  Hashtbl.remove cache domid;
 			  None
 
+  let remove_gone_domains_cache xc =
+	  let current_domains = Xenctrl.domain_getinfolist xc 0 in
+	  Mutex.execute m
+	  (fun () ->
+			try
```
Comment (johnelse) [inline src/squeeze_xen.ml]: Is this `try ... with` needed? I don't think the code it guards can throw an exception.

**Comment by cheng-z** [comment]
Hi @johnelse 
Thanks for your comments. I already modify the code based on your suggestion. Please help to review the code again.
Thanks again.

**Comment by johnelse** [comment]
Thanks!

**Pull request merged**